src/main/java/com/jvm_bloggers/core/rss/BlogPostsController.java

```diff
@@ -1,41 +1,98 @@
 import com.rometools.rome.io.SyndFeedOutput;
-import lombok.RequiredArgsConstructor;
+
 import lombok.SneakyThrows;
+
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Value;
-import org.springframework.http.MediaType;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RequestParam;
 import org.springframework.web.bind.annotation.RestController;
 
 import java.io.PrintWriter;
+import java.util.Arrays;
 import java.util.Set;
 
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
 import static com.google.common.base.MoreObjects.firstNonNull;
+import static com.jvm_bloggers.core.rss.BlogPostsController.SupportedRssFormat.JSON;
+import static com.jvm_bloggers.core.rss.BlogPostsController.SupportedRssFormat.XML;
+import static javaslang.API.$;
+import static javaslang.API.Case;
+import static javaslang.API.Match;
+import static javaslang.API.run;
+import static org.springframework.http.HttpStatus.BAD_REQUEST;
+import static org.springframework.http.MediaType.APPLICATION_ATOM_XML_VALUE;
+import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
 
 @RestController
-@RequiredArgsConstructor(onConstructor = @__(@Autowired))
 public class BlogPostsController {
+    public static final String RSS_FEED_MAPPING = "/pl/rss";
 
     private final AggregatedRssFeedProducer rssProducer;
+    private final SyndFeedToJsonConverter syndFeedToJsonConverter;
+
+    private final Integer defaultLimit;
+
+    @Autowired
+    public BlogPostsController(AggregatedRssFeedProducer rssProducer,
+                               SyndFeedToJsonConverter syndFeedToJsonConverter,
+                               @Value("${generated.rss.entries.limit}") Integer defaultLimit) {
+        this.rssProducer = rssProducer;
+        this.syndFeedToJsonConverter = syndFeedToJsonConverter;
+        this.defaultLimit = defaultLimit;
+    }
 
     @SneakyThrows
-    @RequestMapping("/pl/rss")
+    @RequestMapping(RSS_FEED_MAPPING)
     public void getRss(HttpServletRequest request, HttpServletResponse response,
-        PrintWriter writer, @RequestParam(required = false) Integer limit,
-        @Value("${generated.rss.entries.limit}") int defaultLimit,
-        @RequestParam(required = false) Set<String> excludedAuthors) {
-        limit = firstNonNull(limit, defaultLimit);
-        response.setContentType(MediaType.APPLICATION_ATOM_XML_VALUE);
-        new SyndFeedOutput().output(
-            rssProducer.getRss(request.getRequestURL().toString(), limit, excludedAuthors),
-            writer
+                       PrintWriter writer,
+                       @RequestParam(required = false) Integer limit,
+                       @RequestParam(required = false) Set<String> excludedAuthors,
+                       @RequestParam(required = false, defaultValue = "xml") String format) {
```

Comment:
Current approach requires non standard parameter. Standard approach would be to use HTTP Content Negotiation (Accept header). I think that could be easily achieved by:

1. `@RequestMapping(path = RSS_FEED_MAPPING, produces = {MediaType.APPLICATION_JSON_UTF8_VALUE, MediaType.APPLICATION_ATOM_XML_VALUE})`
2. Returning `SyndFeed` from this method
3. Creating two beans of type `HttpMessageConverter` that would be responsible for serializing XML and JSON responses 

That way you would reuse existing framework mechanism and conform to standard HTTP content negotiation mechanism